Code/Framework/AzFramework/AzFramework/Terrain/TerrainDataRequestBus.h

```diff
@@ -131,6 +134,41 @@ namespace AzFramework
                 Sampler sampleFilter = Sampler::DEFAULT,
                 bool* terrainExistsPtr = nullptr) const = 0;
 
+            //! Given a list of XY coordinates, call the provided callback function with surface data corresponding to each
+            //! XY coordinate in the list.
+            virtual void ProcessHeightsFromList(const AZStd::array_view<AZ::Vector3>& inPositions,
```

Comment:
Should these all accept a list of AZ::Vector2 instead of AZ::Vector3?

I see above that there are single value versions for both Vector2 and Vector3 which makes sense for convenience, but from the perspective of someone using these new API's I think it's better to have them pass in a list of Vector2 instead of Vector3 if the z component is always going to be ignored.

Reply:
Yeah. Vector2 should work here. I'll change.

Reply:
It might be beneficial to have both versions.  The reason for the Vector3 versions is that the gameplay-based use cases for calling this generally come from having positions of various entities that you're trying to get terrain heights for.  You _can_ take each one and stuff it into a Vector2, but it's easier to just pass it in as Vector3s.  (Try the Vector3 -> Vector2 conversions in Script Canvas to see what I mean)

Reply:
I put back the Vector3 version and changed the name of the Vector2 versions to `Process*ListOfVector2` to follow the same pattern as the Get* functions.